### ISSUE TITLE
DocumentDB global cluster: error when creating cross-region replica cluster when primary cluster encrypted

### ISSUE THREAD

**Issue by gherrin-chwy** [hashicorp/terraform-provider-aws#21764]
Unable to create encrypted global replica cluster using KMS keys in each region.  This capability is supported for Aurora global clusters but does not appear to be supported for DocumentDB global cluster.

Error output (names masked):

Error: error creating DocDB cluster: InvalidParameterCombination: Source cluster arn:aws:rds:us-east-1:243249644484:cluster:****************** is encrypted; pre-signed URL has to be specified
        status code: 400, request id: cd36f08b-796f-4d4a-835f-b1191f4dd190

I was able to manually create the cross-region replica cluster using the AWS Console and our customer-managed KMS key for the replica region, but not through Terraform.  

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

Terraform v0.13.7
+ provider registry.terraform.io/hashicorp/aws v3.65.0
+ provider registry.terraform.io/hashicorp/null v3.1.0
+ provider registry.terraform.io/hashicorp/random v3.1.0

Your version of Terraform is out of date! The latest version
is 1.0.11. You can update by downloading from https://www.terraform.io/downloads.html

### Affected Resource(s)

* aws_docdb_cluster

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
resource "aws_docdb_cluster" "secondary" {
  global_cluster_identifier       = var.global_cluster_identifier
  cluster_identifier              = "${local.resource_short_name}-cluster"
  engine                          = "docdb"
  engine_version                  = var.engine_version
  kms_key_id                      = var.kms_db_key_arn
  final_snapshot_identifier       = "${local.resource_short_name}-finalsnapshot-${random_uuid.last_snap_id.result}"
  skip_final_snapshot             = var.skip_final_snapshot
  deletion_protection             = var.deletion_protection
  backup_retention_period         = var.backup_retention_period
  preferred_backup_window         = var.preferred_backup_window
  preferred_maintenance_window    = var.preferred_maintenance_window
  db_subnet_group_name            = local.db_subnet_group_name
  vpc_security_group_ids          = compact(concat(module.security_group.id, var.vpc_security_group_ids))
  snapshot_identifier             = var.snapshot_identifier
  storage_encrypted               = true
  port                            = 27017
  apply_immediately               = var.apply_immediately
  db_cluster_parameter_group_name = aws_docdb_cluster_parameter_group.docdb_cluster_param_group.name
  enabled_cloudwatch_logs_exports = (var.export_logs ? ["audit", "profiler"] : [])

  timeouts {
    create = "120m"
    delete = "120m"
    update = "120m"
  }
  tags = module.core_info.required_tags

  lifecycle {
    ignore_changes = [global_cluster_identifier]
  }
}

```

### Expected Behavior

Expected cross-region replica cluster to be created from encrypted primary cluster and using the appropriate KMS key to encrypted the replica cluster.

### Actual Behavior

Provisioning of aws_docdb_cluster failed.  

Error output (names masked):

Error: error creating DocDB cluster: InvalidParameterCombination: Source cluster arn:aws:rds:us-east-1:243249644484:cluster:****************** is encrypted; pre-signed URL has to be specified
        status code: 400, request id: cd36f08b-796f-4d4a-835f-b1191f4dd190

### Steps to Reproduce

1. Provision regional aws_docdb_cluster resource with storage encrypted by KMS key for that region
2. Provision global cluster, providing source_db_cluster_identifier attribute set to the cluster identifier created in step 1
3. Provision replica cluster, providing global_cluster_identifier from Step 2 AND kms_key_id for KMS key in replica region

### References

* #0000

**Comment by justinretzolk** [comment]
Hey @gherrin-chwy 👋 Thank you for taking the time to file this issue. Something that I was unclear on from your reproduction notes was whether the `aws_docdb_global_cluster` resource has the [`storage_encrypted` parameter](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/docdb_global_cluster#storage_encrypted) set to `true`, which I believe is required in this configuration. Can you confirm whether that parameter has been set?

**Comment by gherrin-chwy** [comment]
@justinretzolk, thanks for your reply.  Yes, storage_encrypted is true on the primary region cluster, which in turn is inherited by the global cluster when I create the global, passing the source_db_cluster_identifier to the "aws_docdb_global_cluster" resource.

**Comment by gherrin-chwy** [comment]
@justinretzolk, I have an update for you on this issue.  It seems to be working properly now, and I think the reason for the issue may have been a race condition.  Our Terraform configuration allows for single-region, non-global cluster creation or global, multi-region clusters.  As such, we have two modules we coded, one for primary-cluster and another for replica-cluster, the latter of which can be called once for each region.  The replica-cluster module is passed the global_cluster_identifier of course.

I believe the error I was receiving may have been due to Terraform attempting to create the replica cluster concurrently with the primary cluster; therefore, I added a module depends_on block to make the replica cluster wait for primary cluster to be completed.  

I believe this issue can be closed.
